osu.Game/OsuGameBase.cs

```diff
@@ -225,6 +228,9 @@ List<ScoreInfo> getBeatmapScores(BeatmapSetInfo set)
             dependencies.Cache(difficultyManager);
             AddInternal(difficultyManager);
 
+            dependencies.Cache(CollectionManager = new BeatmapCollectionManager(Storage));
+            AddInternal(CollectionManager);
```

Comment:
This is adding an extra background thread on startup (`loadDatabase()`), which I'm not sure we want. Any reason this is done in `GameBase` and not `Game`? Just to stay in line with the other "manager"s?

Reply:
Mostly to stay in line with other managers, yes. I think it can be moved.

Reply:
In hindsight it's probably fine to not spawn the extra thread - I mostly did it as a safety insurance because it's querying EF, but that should be fast anyway since it's querying an indexed column...

Reply:
I definitely think it should be threaded, because if someone has a 100mb collection database it shouldn't delay visual startup. Probably something that could be in the `loadComponentSingleFile` queue?

Reply:
Yeah, that'd also be fine.